DLaB.CrmSvcUtilExtensions/NamingService.cs

```diff
@@ -348,13 +349,17 @@ private string AppendValueForDuplicateOptionSetValueNames(OptionSetMetadataBase
         /// <returns></returns>
         public string GetNameForAttribute(EntityMetadata entityMetadata, AttributeMetadata attributeMetadata, IServiceProvider services)
         {
-            return GetNameForAttribute(entityMetadata, attributeMetadata, services, CamelCaseMemberNames);
+            return GetNameForAttribute(entityMetadata, attributeMetadata, services, CamelCaseMemberNames, UseLogicalNames);
         }
 
-        private string GetNameForAttribute(EntityMetadata entityMetadata, AttributeMetadata attributeMetadata, IServiceProvider services, bool camelCase)
+        private string GetNameForAttribute(EntityMetadata entityMetadata, AttributeMetadata attributeMetadata, IServiceProvider services, bool camelCase, bool useLogicalNames)
         {
             string attributeName;
-            if (EntityAttributeSpecifiedNames.TryGetValue(entityMetadata.LogicalName.ToLower(), out var specifiedNames) &&
+            if (useLogicalNames)
+            {
+                attributeName = attributeMetadata.LogicalName;
+            }
+            else if (EntityAttributeSpecifiedNames.TryGetValue(entityMetadata.LogicalName.ToLower(), out var specifiedNames) &&
```

Comment:
This should really be taken priority.  This would then allow someone to be able to specify a different name for an attribute that is the same as the entity name,

Reply:
Changes have been made in the code to prioritize the custom attribute capitalizations over any other naming option. The description of the new setting has been updated too to give more information for the user on how to resolve the issue if it occurs.

Thank you for the review!